std-bits/base/src/main/java/org/enso/base/spi/EnsoService.java

```diff
@@ -0,0 +1,65 @@
+package org.enso.base.spi;
+
+import java.util.logging.Logger;
+import org.enso.base.polyglot.EnsoMeta;
+import org.graalvm.polyglot.PolyglotException;
+import org.graalvm.polyglot.Value;
+
+/** A base class for an Enso service backed by an Enso type. */
+public abstract class EnsoService {
```

Comment:
I would prefer to make this `EnsoServiceLoader.Provider` inner class, but I can live with outer class as well.

Reply:
I don't think this is a provider.

Our Loader scheme omits the provider abstraction as it seems unnecessary in this setup. Note that all the `ServiceLoader.Provider` does is have a `get` method that allows to extract the underlying type `T` (from `ServiceLoader<T>`). In our case, `EnsoService` is the base type for "that `T`" - i.e. it is the bound in `EnsoServiceLoader<? extends EnsoService>`. So they server a bit different purposes and I don't think the name `Provider` would be fitting.